build.install4j

```diff
@@ -334,6 +334,177 @@
+    if(System.getenv("XDG_DATA_HOME") != null || (shareFolder.exists() &amp;&amp; shareFolder.isDirectory())){
+        File applicationsDir = new File(shareFolder, "applications");
+        if(!applicationsDir.exists() || !applicationsDir.isDirectory()){
+            applicationsDir.mkdirs();
+        }
+        File desktopFile = Arrays.stream(applicationsDir.listFiles(new FilenameFilter(){
+            @Override
+            public boolean accept(File dir, String name){
+                return name.matches("TripleA[\\-0-9\\.]*\\.desktop");
+            }
+        })).findAny().get();
+        File mimeappsFile = new File(applicationsDir, "mimeapps.list");
+        if(!desktopFile.exists()){
+            return false;
+        }
+        if(!mimeappsFile.exists()){
+            mimeappsFile.createNewFile();
+        }
+        String mimeType = "x-scheme-handler/triplea";
+        StringBuilder builder = new StringBuilder();
+        try(Scanner scanner = new Scanner(desktopFile)){
+            String mimeTypePrefix = "MimeType=";
+            while(scanner.hasNextLine()){
+                String nextLine = scanner.nextLine();
+                if(nextLine.startsWith("Exec=") &amp;&amp; !nextLine.contains("%u")){
+                    builder.append(nextLine).append(" \"%u\"\n");
+                } else if(!nextLine.startsWith(mimeTypePrefix)){
+                    builder.append(nextLine).append('\n');
+                }
+            }
+            builder.append(mimeTypePrefix).append(mimeType);
+        }
+        try(PrintWriter writer = new PrintWriter(desktopFile)){
+            writer.println(builder.toString());
+        }
+        builder = new StringBuilder();
+        try(Scanner scanner = new Scanner(mimeappsFile)){
+            boolean isSetup = false;
+            String defaultApplicationString = "[Default Applications]";
+            String mimeTypePrefix = mimeType + "=";
+            while(scanner.hasNextLine()){
+                String nextLine = scanner.nextLine();
+                if(!nextLine.startsWith(mimeTypePrefix)){
+                    builder.append(nextLine).append('\n');
+                }
+                if(nextLine.equalsIgnoreCase(defaultApplicationString)){
+                    isSetup = true;
+                }
+            }
+            if(!isSetup){
+                builder.insert(0, defaultApplicationString + "\n");
+            }
+            builder.append(mimeTypePrefix).append(desktopFile.getName());
+        }
+        try(PrintWriter writer = new PrintWriter(mimeappsFile)){
+            writer.println(builder.toString());
+        }
+    }
+}
+return true;</string>
+                              </void>
+                            </object>
+                          </void>
+                        </object>
+                      </java>
+                    </serializedBean>
+                    <condition />
+                  </action>
+                </beans>
+              </group>
```

Comment:
A general question about the new Linux install/uninstall actions...  Will they handle the case where I install multiple versions of TripleA side-by-side?  I don't know about others, but I do this regularly.

My expectation would be that the `triplea` protocol handler would always point to the latest version.  For example:

* If I have version 1 already installed and then install version 2 to a different folder, I would expect the protocol handler to be updated to version 2.
* If I have version 2 already installed and then install version 1 to a different folder (say to try to reproduce a bug), I would expect the protocol handler to remain unchanged.

Note that this could severely complicate the uninstall logic if it has to restore the protocol handlers to point to the previous version instead of just removing them.

What are your thoughts on this, @RoiEXLab?  Are side-by-side installations something we need to worry about for a general user?